### ISSUE TITLE
Using .then for chaining is problematic

### ISSUE THREAD

**Issue by Huulivoide** [finnair/v-validation#40]
Using **then** as the function name for chaining is problematic.

Objects implementing a function named **then** aka _thenable objects_ already have predefined functionality.
Thenable objects behave like Promises to an extends. For example if you pass thenable to Promise.resolve(), it will assume that the thenable is promise-like and will wrap it in a real promise that will wait for the thenable to resolve or reject.
Also the same things happens if the validator is returned from manually created Promise with resolve, or returned in the middle of promise .then call chain.

This now has the implication that it is impossible to return the validators directly from async functions, as they perform (or atleast behave similarly to) Promise.resolve in then return value.

Example in which the validator is never returned as the async function tries to wait for the NumberNormalizer's then method to resolve or reject.

```javascript
const { V } = require('@finnair/v-validation');

let validatorCreated = false;

async function createValidator() {
  return V.toNumber();
}

createValidator()
  .then(v => {
    console.log('Then:', v);
    validatorCreated = true;
  })
  .catch(e => console.error('Error:', e));

function loop() {
  console.log('validatorCreated =', validatorCreated);
  
  if (!validatorCreated) {
    setTimeout(loop, 1000);
  }
}

loop();
```

**Comment by ssaarela** [comment]
For us `then` felt like the most natural way of describing validator chains and it was never a problem as the context of defining validation rules is different from actual promise chaining (i.e. validating). Also TypeScript doesn't allow that kind of call as `Validator.then` requires a `Validator` as a parameter. However, I can see this being a bit problematic choice of naming for plain JavaScript. 

I think `then` and `thenMap` could be refactored to e.g. `next` and `nextMap` without affecting readability of the rules too much 🤔  How does that sound? Some other, better name? 

I'd also like to hear opinions of other `v-validation` users/contributors about this. Ping @paazmaya , @eostman , @N1cc3 and @Jsemi what do you think?

**Comment by paazmaya** [comment]
Renaming to `next*` sounds like a good solution to go forward

**Comment by ssaarela** [comment]
This is now published as part of 0.7.0 release.
